internal/backend/grpc/management.go

```diff
@@ -291,6 +345,47 @@ func (s *managementServer) adminTalosconfig(ctx context.Context) (*management.Ta
 	}, nil
 }
 
+func (s *managementServer) adminKubeconfig(ctx context.Context) (*management.KubeconfigResponse, error) {
+	_, err := auth.CheckGRPC(ctx, auth.WithRole(role.Admin))
+	if err != nil {
+		return nil, err
+	}
+
+	if !constants.IsDebugBuild || !config.Config.EnableBreakGlassConfigs {
+		return nil, status.Error(codes.PermissionDenied, "not allowed")
+	}
+
+	routerContext := router.ExtractContext(ctx)
+
+	if routerContext == nil || routerContext.Name == "" {
+		return nil, status.Error(codes.InvalidArgument, "cluster name is required")
+	}
+
+	clusterName := routerContext.Name
+
+	type kubernetesAdmin interface {
+		AdminKubeconfig(ctx context.Context, id string) ([]byte, error)
+	}
+
+	kubernetesRuntime, err := runtime.LookupInterface[kubernetesAdmin](kubernetes.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := kubernetesRuntime.AdminKubeconfig(ctx, clusterName)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = s.markClusterAsTainted(ctx, clusterName); err != nil {
```

Comment:
Cluster is a user-owned resource, so they could remove that taint? Should we use another resource?

Reply:
Applying a cluster template would remove it as well.

Reply:
I think `ClusterStatus` is the right place, but why copy? Just write on it?

Reply:
It's owned by the controller, didn't want to interfere that

Reply:
Oh, you are right. Copying would be kind of ok, my only concern would be, when I sync my cluster template next time, it would report as changed, when actually nothing should be changed.

Would adding another resource be an overkill? Is there any other existing ownerless resource we can use?

Reply:
Well, not a big deal. Can add a resource too. But it will be possible to destroy this resource too unless we set an owner

Reply:
Ah wait nvm, we can block write access there.